server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationIT.java

```diff
@@ -1622,4 +1627,170 @@ public void testRealtimeMultiGetRequestsUnsuccessful() {
         assertTrue(mgetResponse.getResponses()[1].isFailed());
 
     }
+
+    /**
+     * Tests whether segment replication supports realtime termvector requests and reads and parses source from the translog to serve strong reads.
+     */
+    public void testRealtimeTermVectorRequestsSuccessful() throws IOException {
+        final String primary = internalCluster().startDataOnlyNode();
+        XContentBuilder mapping = jsonBuilder().startObject()
+            .startObject("properties")
+            .startObject("field")
+            .field("type", "text")
+            .field("term_vector", "with_positions_offsets_payloads")
+            .field("analyzer", "tv_test")
+            .endObject()
+            .endObject()
+            .endObject();
+        // refresh interval disabled to ensure refresh rate of index (when data is ready for search) doesn't affect realtime termvectors
+        assertAcked(
+            prepareCreate(INDEX_NAME).setMapping(mapping)
+                .addAlias(new Alias("alias"))
+                .setSettings(
+                    Settings.builder()
+                        .put(indexSettings())
+                        .put("index.analysis.analyzer.tv_test.tokenizer", "standard")
+                        .put("index.refresh_interval", -1)
+                        .putList("index.analysis.analyzer.tv_test.filter", "lowercase")
+                )
+        );
+        final String replica = internalCluster().startDataOnlyNode();
+        ensureGreen(INDEX_NAME);
+        final String id = routingKeyForShard(INDEX_NAME, 0);
+
+        TermVectorsResponse response = client(replica).prepareTermVectors(indexOrAlias(), "1").get();
+        assertFalse(response.isExists());
+
+        // index doc 1
+        client().prepareIndex(INDEX_NAME)
+            .setId(Integer.toString(1))
+            .setSource(
+                jsonBuilder().startObject()
+                    .field("field", "the quick brown fox jumps over the lazy dog")
+                    // 0the3 4quick9 10brown15 16fox19 20jumps25 26over30
+                    // 31the34 35lazy39 40dog43
+                    .endObject()
+            )
+            .execute()
+            .actionGet();
+
+        // non realtime termvectors 1
+        response = client().prepareTermVectors(indexOrAlias(), Integer.toString(1)).setRealtime(false).get();
+        assertFalse(response.isExists());
+
+        // realtime termvectors 1
+        TermVectorsRequestBuilder resp = client().prepareTermVectors(indexOrAlias(), Integer.toString(1))
```

Comment:
Nit: mention `.setRealtime(true)` to be explicit about creating a realtime request